poly-commit/src/linear_codes/mod.rs

```diff
@@ -339,37 +335,24 @@ where
             // 3. Generate vector `b` to left-multiply the matrix.
             let (_, b) = L::tensor(point, n_cols, n_rows);
 
-            let mut transcript = IOPTranscript::new(b"transcript");
-            transcript
-                .append_serializable_element(b"root", root)
-                .map_err(|_| Error::TranscriptError)?;
+            sponge.absorb(&to_bytes!(&commitment.root).map_err(|_| Error::TranscriptError)?);
 
             // If we are checking well-formedness, we need to compute the well-formedness proof (which is just r.M) and append it to the transcript.
             let well_formedness = if ck.check_well_formedness() {
-                let mut r = Vec::new();
-                for _ in 0..n_rows {
-                    r.push(
-                        transcript
-                            .get_and_append_challenge(b"r")
-                            .map_err(|_| Error::TranscriptError)?,
-                    );
-                }
+                // TODO Check this: we changed, now we squeeze all elements and then abosrb all of them
+                let r = sponge.squeeze_field_elements::<F>(n_rows);
+                sponge.absorb(&r);
```

Comment:
why are we doing this, actually? I'm no expert on building the IOP transcripts, but it seems counterintuitive to reabsorb whatever we just squeezed. Maybe @mmaker would know if it would be safe to just remove that?

Reply:
I don't know either. I just imitated `get_and_append_challenge` here.